src/main/java/htsjdk/samtools/sra/SRAAlignmentIterator.java

```diff
@@ -191,4 +196,14 @@ private void nextReference() {
 
         return referencesChunks;
     }
+
+    @Override
+    public void close() {
+        if (lastRecord != null) {
+            lastRecord.detachFromIterator();
```

Comment:
why is this important? can you explain? from what I see detach... simply sets some variables to null, given that you are nullifying lastRecord in the next step, what good does it do?